src/components/PeopleFilters.tsx

```diff
@@ -1,21 +1,67 @@
-export const PeopleFilters = () => {
+import { memo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { CENTURIES } from '../utils/constants';
+import { SearchParam, Sex } from '../types';
+import { toggleCentury } from '../utils';
+import { SearchLink } from './SearchLink';
+
+export const PeopleFilters = memo(() => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const sex = searchParams.get(SearchParam.Sex) ?? '';
+  const query = searchParams.get(SearchParam.Query) ?? '';
+  const centuries = searchParams.getAll(SearchParam.Centuries) ?? [];
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const params = new URLSearchParams(searchParams);
+    const inputValue = event.target.value;
+
+    if (inputValue) {
+      params.set(SearchParam.Query, inputValue);
+    } else {
+      params.delete(SearchParam.Query);
+    }
+
+    setSearchParams(params);
```

Comment:
You can get and modify previous search params with setSearchParams.
Something like this:
```javascript
setSearchParams((prevParams) => {
  if (inputValue) {
    prevParams.set(SearchParam.Query, inputValue);
  } else {
    prevParams.delete(SearchParam.Query);
  }

  return prevParams;
});
```